client/server/silicon/silicon_utils.ts

```diff
@@ -0,0 +1,21 @@
+import { EncryptedMarket, Market } from '../../graphql/graphql';
+import { Silicon } from './silicon'; // Assume this is the path to your Silicon class
+import { poseidonHashMany } from "@scure/starknet";
+
+export function verifySiliconMapping(encrypted_silicon: Silicon<EncryptedMarket>, transparent_silicon: Silicon<Market>): boolean {
+    if (encrypted_silicon.markets.length !== transparent_silicon.markets.length) return false;
+
+    for (let i = 0; i < encrypted_silicon.markets.length; i++) {
+        const encrypted_market = encrypted_silicon.markets[i];
+        const transparent_market = transparent_silicon.markets.find(m => m.game_id === encrypted_market.game_id && m.location_id === encrypted_market.location_id && m.drug_id === encrypted_market.drug_id);
+
+        if (!transparent_market) return false; // No corresponding market found in transparent_silicon
+
+        const hashCash = poseidonHashMany([BigInt(transparent_market.cash)]);
```

Comment:
May need to add a blinding factor here

Or else hash doesn't have 256 bits of security

Reply:
your hash*

Reply:
this is replicating first and sole hash done on-chain when we init the "blinded markets", can go over in call if in dm if you want... so initial "blinding" is mostly cosmetics security wise.

Reply:
understood!